src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -1,22 +1,107 @@
 package io.stargate.sgv2.jsonapi.service.cqldriver.executor;
 
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.data.ByteUtils;
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.core.metadata.schema.IndexMetadata;
 import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import com.datastax.oss.driver.api.core.type.VectorType;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.sgv2.jsonapi.exception.SchemaException;
+import io.stargate.sgv2.jsonapi.service.schema.SimilarityFunction;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 public class TableSchemaObject extends TableBasedSchemaObject {
 
   public static final SchemaObjectType TYPE = SchemaObjectType.TABLE;
 
-  public TableSchemaObject(TableMetadata tableMetadata) {
+  private final VectorConfig vectorConfig;
+
+  private TableSchemaObject(TableMetadata tableMetadata, VectorConfig vectorConfig) {
     super(TYPE, tableMetadata);
+    this.vectorConfig = vectorConfig;
   }
 
   @Override
   public VectorConfig vectorConfig() {
-    return VectorConfig.notEnabledVectorConfig();
+    return vectorConfig;
   }
 
   @Override
   public IndexUsage newIndexUsage() {
     return IndexUsage.NO_OP;
   }
+
+  /**
+   * Get table schema object from table metadata
+   *
+   * @param tableMetadata
+   * @param objectMapper
+   * @return
+   */
+  public static TableSchemaObject getTableSettings(
```

Comment:
Ok so this is a factory method, so name should probably be something like... `from()` or `construct()`? Or whatever names we usually use. "getXxx()" suggested getter.